Fixes check_db to check for the actual block

SWvheerden opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
This fixes the check_db command to ensure that the actual block can be found.
Also adds a block so it doesnt look for a blocks older than the pruning horizon. 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Currently the check_db only checks that the get_blocks return OK, but this will always happen as we cant return an error to another node. This checks that the function actually returns the block as the block we ask for needs to be in the db. 

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
* [x] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

hansieodendaal commented:
Works for me on an unbroken db, logic makes sense, just need to fix clippy warnings.

neonknight64 commented on applications/tari_base_node/src/parser.rs:
```diff
@@ -1140,14 +1140,24 @@ impl Parser {
-                    missing_blocks.push(height);
-                };
+                // we can only check till the pruning horizon, 0 is archive node so it needs to check every block.
+                if height > meta.pruning_horizon {
```
The pruning_horizon should be used to calculate the horizon height using the ChainMetadata horizon_block function. The height should then be larger than the horizon height and not the pruning horizon.

SWvheerden replied:
You're correct, fixing it now

neonknight64 commented on applications/tari_base_node/src/parser.rs:
```diff
@@ -1140,14 +1140,25 @@ impl Parser {
             let mut missing_headers = Vec::new();
             print!("Searching for height: ");
+            // We need to check every header, but not every block.
+            let pruning_horizon = height - meta.pruning_horizon;
```
I believe pruning horizon has a different definition and horizon height or horizon block should rather be used.
```suggestion
            let horizon_height = meta.horizon_block(height);
```

neonknight64 commented on applications/tari_base_node/src/parser.rs:
```diff
@@ -1140,14 +1140,25 @@ impl Parser {
-                    missing_blocks.push(height);
-                };
+                // we can only check till the pruning horizon, 0 is archive node so it needs to check every block.
+                if height > pruning_horizon {
```
```suggestion
                if height > horizon_height {
```

SWvheerden committed:
Fixes check_db to check for the actual block

add pruning horizon check

check every header but not block


Update applications/tari_base_node/src/parser.rs

Co-authored-by: Yuko Roodt <42933690+neonknight64@users.noreply.github.com>
Update applications/tari_base_node/src/parser.rs

Co-authored-by: Yuko Roodt <42933690+neonknight64@users.noreply.github.com>

sdbondi reviewed (approved):
LGTM
